### ISSUE TITLE
 removing specific tag fails / 0x80004005 #76 

### ISSUE THREAD

**Issue by toweih** [RealRaven2000/FiltaQuilla#129]
select mails by a specific tag ("detach"). For these I detach all files and let the tag be removed afterwards, to avoid double-processing. But since the last update, removing the tag fails on TB 78.14.0 (64-Bit) using Quickfilters 5.3.2 on a debian linux.

Logs-Trace:
[03.10.21, 21:54:34 GMT+2] Filter "Anhänge abtrennen" auf die Nachricht *** am 29.09.21, 12:24:24 GMT+2 angewendet Anhänge abtrennen nach

[03.10.21, 21:54:34 GMT+2] Filter "Anhänge abtrennen" auf die Nachricht *** am 29.09.21, 12:24:24 GMT+2 angewendet Schlagwort entfernen

[03.10.21, 21:54:34 GMT+2] Mitteilung von Filter "Anhänge abtrennen": custom action failed to apply

[03.10.21, 21:54:37 GMT+2] Filteraktion fehlgeschlagen: "filterActionFailed" mit Fehlercode 0x80004005 beim Versuch folgender Aktion: Filter "Anhänge abtrennen" auf die Nachricht - am 01.01.70, 01:00:00 GMT+1 angewendet Schlagwort entfernen

[03.10.21, 21:54:37 GMT+2] Filter "Anhänge abtrennen" auf die Nachricht - am 01.01.70, 01:00:00 GMT+1 angewendet Schlagwort hinzugefügt

**Comment by toweih** [comment]
This is my filter:
![image](https://user-images.githubusercontent.com/19357553/139674254-d81d8527-1a9c-4bca-8670-4b895dc1517e.png)

Error appears on TB 78.14.0, FiltaQuilla 3.2

**Comment by RealRaven2000** [comment]
Could you try to do the actual detaching of the Email as last action?

**Comment by toweih** [comment]
No, unfortunatly nothing changes when I run the detach-operation as last step. 
Adding a new label "processed" and adding it as a condition works for me.

**Comment by RealRaven2000** [comment]
still no idea on this - I don't know what the internal function [`messenger.detachAttachmentsWOPrompts`](https://searchfox.org/comm-esr91/source/mailnews/base/src/nsMessenger.cpp#647) (source linked on searchfox.org) does in minute detail - maybe it locks the database so that the tags cannot be overwritten. Maybe it creates a copy of the message and throws the original away (so that the tag changes are applied to a dead object). Not sure how to detect that. Since the adding / removal of tags is also a built in  method of Thunderbird I cannot  control what they do exactyl.
